mythril/analysis/modules/suicide.py

```diff
@@ -54,15 +55,25 @@ def _analyze_state(state, node):
         description += "The remaining Ether is sent to: " + str(to) + "\n"
 
     not_creator_constraints = []
-    if len(state.world_state.transaction_sequence) > 1:
-        creator = state.world_state.transaction_sequence[0].caller
-        for transaction in state.world_state.transaction_sequence[1:]:
-            not_creator_constraints.append(
-                Not(Extract(159, 0, transaction.caller) == Extract(159, 0, creator))
-            )
-            not_creator_constraints.append(
-                Not(Extract(159, 0, transaction.caller) == 0)
-            )
+    if len(state.world_state.transaction_sequence) >= 1:
+        creator = None
+        if isinstance(
+            state.world_state.transaction_sequence[0], ContractCreationTransaction
+        ):
+            creator = state.world_state.transaction_sequence[0].caller
+        if creator is not None:
+            for transaction in state.world_state.transaction_sequence[1:]:
+                not_creator_constraints.append(
+                    Not(Extract(159, 0, transaction.caller) == Extract(159, 0, creator))
+                )
+                not_creator_constraints.append(
+                    Not(Extract(159, 0, transaction.caller) == 0)
+                )
+        else:
+            for transaction in state.world_state.transaction_sequence:
+                not_creator_constraints.append(
+                    Not(Extract(159, 0, transaction.caller) == 0)
```

Comment:
The problem here is that the owner can suicide the contract, and the module won't know

Reply:
I think for on chain analysis the specification should be different, namely something like:
If somebody with a specific address can trigger a sequence of transactions that triggers the suicide then the contract is vulnerable. (We can even make this configurable)

Reply:
Or what previously happened, check if there is a constraint on the caller and check if you can change that constraint

Reply:
previous implementation had a vague check on changeability for caller, should we do something regarding that at this moment?